__tests__/react/clear.js

```diff
@@ -72,14 +78,18 @@ describe("userEvent.clear", () => {
     it.each(["input", "textarea"])(
       `should clear when <%s> is of type="${type}"`,
       (inputType) => {
+        const onChange = jest.fn().mockImplementation((event) => {
+          expect(input).toHaveProperty("value", "");
+          expect(event.target).toHaveProperty("value", "");
```

Comment:
I'm wondering is there a specific case why assertions placed inside a mocked implementation instead of test flow? Isn't `input` strict equal `event.target` here and already checked in the end of the test (on line 100)?

Reply:
Yes, `input` is strict equal to `event.target`.

I ran into an edge case with the implementation that I've since scrapped. The sequence was events was as follows:

* call `input`'s prototype `value` setter
* fired change event (wasn't using `fireEvent.change`)
* use `input.value` setter, which is overridden by React

In this case, when the event handler was fired, `input.value` would still be the previous value because `input.value`'s getter is also overridden by React and uses an internal copy of the state. If we didn't use `input.value = `to assign the value, then the internal state would not be updated. This is super subtle and I'm fine getting rid of it, but this does add a little bit of confidence. I'll add a comment in the meantime and remove if it that's what you want.